### ISSUE TITLE
accessing injected router service in the application route causes init to fire repeatedly

### ISSUE THREAD

**Issue by efx** [emberjs/ember.js#17873]
I expect the `#init` hook of any route to run once and not repeatedly. When I inject the router service into a route and call `this.router.recognize` I see the `#init` hook fire multiple times.
You can see this in the follow application:
1. `git clone https://github.com/efx/poc-bug/ --branch 3.8.1 && cd poc-bug/route-recognize`
2. `yarn start`

Open your console and you should see the `#init` log multiple times. If you remove[ the line ](https://github.com/efx/poc-bug/blob/3.8.1/route-recognize/app/routes/application.js#L9) that calls `this.router.recognize`, the `init` fires once. I verified `3.8.0` and master as of [`3.11.0-canary+7da998b3` ](https://github.com/emberjs/ember.js/commit/7da998b3c28c4cd67c65814c0718dac52252cae0) exhibits the same bug.

**Comment by xg-wang** [comment]
Accessing RouterService in Router will cause circular loop because Router is injected into RouterService https://github.com/emberjs/ember.js/blob/v3.10.0-beta.2/packages/@ember/-internals/routing/lib/services/router.ts#L66. What methods do you need from RouterService during Router init?

**Comment by efx** [comment]
hmm, so this would be a duplicate of https://github.com/emberjs/ember.js/issues/17791?
I needed the `on` method to register handlers for route transitions.

**Comment by efx** [comment]
Verified this still occurs on 3.11. To be clear, I am accessing the router service from within a route not from the router.
@xg-wang honestly I've forgotten what I was trying to do when I opened this issue. Given I unblocked our application we could close this if expected behavior but I do find this very counter-intuitive and would be curious why this happens. ([Updated link of the router service](https://github.com/emberjs/ember.js/blob/master/packages/%40ember/-internals/routing/lib/services/router.ts#L67))
